src/Mod/PartDesign/App/FeatureHole.cpp

```diff
@@ -803,7 +816,7 @@ void Hole::updateDiameterParam()
             }
             // if nothing was found (e.g. if not metric), we must calculate
             if (!found) {
-                diameter = (5 * ((int)((diameter * 110) / 5))) / 100.0;
+                diameter = diameter * 1.1;

```

Comment:
> The new formula will reproduce the old values for whole numbers which is good. However, the formula will not give the same results for fractional values.

I know. But this formula is arbitrary and we only use it when we don't have a normed value, so only for the rather exotic M2.2, M9, M11 and M17. And the rounding error is negligible.

Concerning the clearance holes, the ones for the UTS profiles are not yet implemented.
But this is another issue. I cannot change too many things at once. Meanwhile I got the ASME 18.2.8 where the clearances are defined, I can do this as soon as this PR is in.

Reply:
Ok, old formula gives nicer values. The fractional ones will look strange on a drawing. I guess that was the original idea. And this will break old models. But it is up to you. Surely there is a way to get rid of the compiler warning by explicitly casting back to double. 
Your call! As long as the consequences are understood.

Reply:
> And this will break old models.

There is no breakage, only for one special case the diameter will be slightly different. The only affected size is M2.2 and the change is < 0.6 %. For all other sizes we use the norm or the error is zero (e.g. M17).
However, the rounding was as arbitrary as the formula itself - it lead to wrong values for all UTS holes.

I fixed the missing UTS clearance hole definitions not too. Now we take the values given in the norm ASME B18.2.8. This changes the the dimameters of existing holes of course, but now we do it right.

Reply:
Sounds good. Any dimensional change should of course be done carefully, as we dont know if/how somebody is dependent on this diameter. That being said, if it is wrong we should fix it. Changing to better standard compliance sounds justified in this case.